### PR TITLE
remote_access: Fix TLS priority test for older TLS versions

### PR DESCRIPTION
**Pull request by kamilvarga**
Fix added to cover cases, when the older TLS versions are not supported(from
RHEL9.0). New check added to make sure, the TLS version 1.1 is supported or
the 1.2 version is used for test instead.

Signed-off-by: Kamil Varga <kvarga@redhat.com>

Test result prior the fix:
<pre>avocado run --vt-type libvirt --vt-machine-type q35 virsh.remote_tls_priority.positive_testing
WARNING:root:No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
WARNING:root:No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
<font color="#2A7BDE">JOB ID     : f6a255bec14802dff9cdf6835886629a85c0729e</font>
<font color="#2A7BDE">JOB LOG    : /root/avocado/job-results/job-2021-07-19T11.36-f6a255b/job.log</font>
 (1/1) type_specific.io-github-autotest-libvirt.virsh.remote_tls_priority.positive_testing: <font color="#F66151">FAIL: TLS priorities test failed for case when the client supports TLSv1.0 only by URI and the server does not support SSLv3 only.</font> (89.94 s)
<font color="#2A7BDE">RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0</font>
<font color="#2A7BDE">JOB TIME   : 90.41 s</font>
</pre>

Test result after the fix is applied:
<pre>avocado run --vt-type libvirt --vt-machine-type q35 virsh.remote_tls_priority.positive_testing
WARNING:root:No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
WARNING:root:No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
<font color="#2A7BDE">JOB ID     : f201a8239f989e0cf5b2311ce064ef6732209b0e</font>
<font color="#2A7BDE">JOB LOG    : /root/avocado/job-results/job-2021-07-20T07.53-f201a82/job.log</font>
 (1/1) type_specific.io-github-autotest-libvirt.virsh.remote_tls_priority.positive_testing: <font color="#33DA7A">PASS</font> (106.78 s)
<font color="#2A7BDE">RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0</font>
<font color="#2A7BDE">JOB TIME   : 107.34 s</font>
</pre>

### CODE REVIEW

**Review by iccaszhulili — APPROVED** [review]

**Comment by iccaszhulili** [comment]
LGTM

**Review by Yingshun — APPROVED** [review]

File: libvirt/tests/src/remote_access/remote_tls_priority.py [hunk 1]
```diff
@@ -134,6 +135,16 @@ def run(test, params, env):
     firewall_cmd.add_port(tls_port, 'tcp', permanent=True)
     server_session.close()
 
+    # Check which version to use for TLS v1.X
+    ssl_check = process.run('openssl s_client -connect {}:16514 -tls1_1'.
+                            format(server_ip), ignore_status=True)
+    if ssl_check.exit_status:
+        logging.debug('TLSv1.1 not supported, use TLS v1.2 instead')
```
Comment (chunfuwen) [inline libvirt/tests/src/remote_access/remote_tls_priority.py:142]: change:
LSv1.1 not supported --> LSv1.0 not supported ?

Comment (kamilvarga) [reply libvirt/tests/src/remote_access/remote_tls_priority.py:142]: We can add check for all versions, but the change in the RHEL 9 is deprecation of the TLS 1.0 and 1.1. In general, the versions below 1.1 are suggested to be deprecated due to security and therefore, the check for TLS1.1 seems to be enough from my point of view. What do you think?

**Review by chunfuwen — APPROVED** [review]
LGTM

**Pull request merged**